tests/core/contracts/test_extracting_event_data.py

```diff
@@ -573,8 +588,13 @@ def test_event_rich_log(
         expected_args):
 
     emitter_fn = emitter.functions[contract_fn]
-    event_id = getattr(emitter_event_ids, event_name)
-    txn_hash = emitter_fn(event_id, *call_args).transact()
+    if hasattr(emitter_event_ids, event_name):
+        event_id = getattr(emitter_event_ids, event_name)
+        txn_hash = emitter_fn(event_id, *call_args).transact()
+    else:
+        # Some tests do not rely on the event_id. Rather than changing this test too much,
```

Comment:
Is this because some of the events aren't in the enum? If it is, I'd lean toward just adding those events to the enum instead of this logic 🤷‍♀️ If it's not, feel free to ignore this comment :)

Reply:
It's more so because of the `if` / `else` logic related to some methods in the `emitter` contract. The "emitter event id" happens to be the first argument passed to some of those methods and then different events are emitted based on which `id` is passed in.

But some methods don't take the `id` as a first argument, just the args related to the method and event itself, and thus would break in this test.

e.g. when the `id` is used:
``` python
function logDouble(WhichEvent which, uint arg0, uint arg1) public {
        if (which == WhichEvent.LogDoubleArg) emit LogDoubleArg(arg0, arg1);
        else if (which == WhichEvent.LogDoubleWithIndex) emit LogDoubleWithIndex(arg0, arg1);
        else if (which == WhichEvent.LogDoubleAnonymous) emit LogDoubleAnonymous(arg0, arg1);
        else revert("Didn't match any allowable event index");
}
```